Automattic/jetpack
# Add post-purchase onboardings to Recommendation flows

This PR **adds post-purchase onboardings to the Recommendations assistant** - it is being triggered for the most relevant purchase (out of supported ones - listed below) and overwrites the normal recommendations steps until it's completed.

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/8419292/193134881-1225bcd4-dcf2-4e93-a371-26ffb63308a7.png">

**Supported purchases:**
- Jetpack Complete
- Jetpack Security
- Jetpack Backup
- Jetpack Anti-Spam
- Jetpack VideoPress
- Jetpack Search
- Jetpack Scan





#### Changes proposed in this Pull Request:

* Adds post-purchase onboarding mechanism
* Adds new assets together with new recommendations cards
* Adds new component (<PrimarySummary />) and modifies existing ones to support new onboardings

#### Other information:


#### Jetpack product discussion


Project thread: pbNhbs-3dq-p2

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:
* Remove all purchases connected to your test site (via Network Admin MC)
* Checkout this PR (e.g. `gh pr checkout 26484`)
* Go to WP Admin and `https://jetpack.com/redirect/?source=jetpack-plans&site=<your_test_site>`
* Purchase first of plans: Jetpack Scan (or Jetpack Complete that is the most complete onboarding - but will require you to remove the purchase in order to test next onboardings)
* You should be redirected to the recommendations page with a `Welcome to __!` card. E.g.:
<img width="1011" alt="image" src="https://user-images.githubusercontent.com/8419292/193134881-1225bcd4-dcf2-4e93-a371-26ffb63308a7.png">

* You should be able to go through the process, and all links should direct you to the valid pages on Jetpack Cloud or WP Admin dashboards
* The summary should include a summary of all features you've been presented. E.g.:

<img width="1003" alt="image" src="https://user-images.githubusercontent.com/8419292/193137841-f2069174-0309-4be4-8ff6-84a431706c29.png">

* Ensure all flows correspond to designs: p1HpG7-hqv-p2

* Best test plan: repeat the steps for all of 7 purchases in this order:

1. Jetpack Scan
2. Jetpack Search
3. Jetpack VideoPress
4. Jetpack AntiSpam
5. Jetpack Backup
6. Jetpack Security
7. Jetpack Complete

* Additionally: Try to purchase Jetpack Boost and ensure it doesn't break the recommendations flow and will not trigger any onboarding.

Every purchase made in this order should overwrite the previous onboarding. And if you already have e.g. Jetpack Complete, the purchase of the "lower" product should show you the onboarding process, but do not impact the summary on the last step - this should depend on the most relevant purchase.
